units/host_monitoring_idle_termination.go

```diff
@@ -174,12 +172,15 @@ func (j *idleHostJob) checkAndTerminateHost(ctx context.Context, schedulerConfig
 	if h.RunningTaskGroup != "" {
 		idleThreshold = idleThreshold * 2
 	} else if hasOutdatedAMI {
-		idleThreshold = outdatedIdleTimeCutoff
+		// Since tasks created after the AMI is updated will only run on new hosts,
+		// we want to terminate outdated hosts aggressively to ensure we're respecting task priorities.
+		idleThreshold = 0
 	}
 
-	// if we haven't heard from the host or it's been idle for longer than the cutoff, we should terminate
 	var terminateReason string
-	if hasOutdatedAMI {
+	if hasOutdatedAMI && communicationTime >= idleThreshold {
```

Comment:
Is `communicationTime >= idleThreshold` redundant, since `idleThreshold` will always be 0 if `hasOutdatedAMI` is true?